subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

```diff
@@ -38,6 +38,20 @@ Some plugins will break with this new version of Gradle, for example because the
 
 === Potential breaking changes
 
+==== Removal of `compile` and `runtime` configurations
+
+Since its inception, Gradle provided the `compile` and `runtime` configurations to declare dependencies.
+Alternatives were introduced in Gradle 3.4:
```

Comment:
Interesting that you introduced "better replacements" here. I had written something very similar in the beginning but I felt like it was "academic arguments" which didn't help if you are frustrated that the configurations are removed. In other words I chose to focus less on the "why" and more on the "how to fix".

Reply:
I think exactly because this is a disruptive change for some builds (which would be the reason to be "frustrated") it is good to give a short idea why this is done in one sentence. If this would be done without reason, that would be really frustrating.

I don't think the argument is academic. We explain below directly, why this is better.